File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> res = new ArrayList<>(candies.length);
        int max = 0;
        for(int candy : candies) {
            max = Math.max(max, candy);
        }

        for(int candy: candies) {
            if (candy + extraCandies >= max)
                res.add(true);
            else
                res.add(false);
        }
        return res;
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.ts
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
    const max: number = Math.max(...candies);
    const rtn : boolean[] = [];

    for(const candy of candies)
        rtn.push(candy + extraCandies >= max);

    return rtn;
};